ecommerce/core/templatetags/core_extras.py

```diff
@@ -31,6 +36,90 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(name='is_special_coupon')
+def is_special_coupon(object_passed):
+    """
+    Checks if the coupon name meets the format criteria of a special coupon. The coupon name is
+    extracted from the 'object_passed' argument which could be an Order, Basket or OrderDiscount instance.
+
+    Arguments:
+        object_passed (Order or SimpleLazyObject or OrderDiscount): Order of the purchase, its basket or its OrderDiscount.
+
+    Returns:
+        bool: True if the coupon name is a valid special coupon, False otherwise.
+    """
+    is_order = isinstance(object_passed, Order)
+    is_basket = isinstance(object_passed, SimpleLazyObject)
+    is_order_discount = isinstance(object_passed, OrderDiscount)
+    coupon_name = ''
+
+    if not is_order and not is_basket and not is_order_discount:
+        return False
+
+    if is_order:
+        discount = object_passed.basket_discounts.first()
+        coupon_name = discount.voucher.name if discount else ''
+
+    if is_basket:
+        coupon = object_passed.vouchers.first()
+        coupon_name = coupon.name if coupon else ''
+
+    if is_order_discount:
+        coupon_name = object_passed.voucher.name
+
+    if not coupon_name:
+        return False
+
+    request = get_current_request()
+    prefix = request.site.siteconfiguration.custom_settings.get('REMOVE_SPECIAL_COUPON_OFFER_PREFIX', '')
+    is_valid_special_coupon = re.match(
+        r'[{prefix}]\d.\d'.format(prefix=prefix),
+        coupon_name,
+    )
+    if not is_valid_special_coupon:
+        return False
+
+    return True
+
+
+@register.filter(name='get_special_coupon_data')
+def get_special_coupon_data(object_passed):
+    """
+    Retrives a custom message for the special coupon.
+
+    Arguments:
+        object_passed (Order or SimpleLazyObject): Order of the purchase or its basket.
+
+    Returns:
+        str: Special coupon message.
+    """
+    is_order = isinstance(object_passed, Order)
+    is_basket = isinstance(object_passed, SimpleLazyObject)
+    coupon_name = ''
+
+    if not is_order and not is_basket:
+        return ''
+
+    if is_order:
+        discount = object_passed.basket_discounts.first()
+        coupon_name = discount.voucher.name if discount else ''
+
+    if is_basket:
+        coupon = object_passed.vouchers.first()
+        coupon_name = coupon.name if coupon else ''
+
+    if not coupon_name:
+        return ''
+
+    request = get_current_request()
+
+    return '{} {}/{}'.format(
+        request.site.siteconfiguration.custom_settings.get('SPECIAL_COUPON_MESSAGE', ''),
+        coupon_name[1],
+        coupon_name[3],
```

Comment:
To avoid exceptions, catch an IndexError here. @Jacatove

Reply:
I added the function `is_valid_special_coupon` which makes sure those indices exist. @Squirrel18